app/src/main/java/ghcharactertracker/com/ghcharactertracker/DBHandler.java

```diff
@@ -0,0 +1,159 @@
+package ghcharactertracker.com.ghcharactertracker;
+
+import android.content.ContentValues;
+import android.content.Context;
+import android.database.Cursor;
+import android.database.sqlite.SQLiteDatabase;
+
+import java.lang.reflect.Array;
+import java.util.ArrayList;
+
+/**
+ * Created by jesse.mailhot on 3/20/2018.
+ */
+
+public class DBHandler {
+    private static DBHandler dbHandler;
+    private SQLiteDatabase db;
+
+    private DBHandler(Context context) {
+        db = context.openOrCreateDatabase("DATABASE", Context.MODE_PRIVATE, null);
+
+        db.execSQL(PlayerContract.PlayerEntry.OPEN_OR_CREATE_CHARACTERS);
+        db.execSQL(PlayerContract.ScenarioEntry.OPEN_OR_CREATE_SCENARIOS);
+    }
+
+    public static DBHandler getDbHandler(Context context) {
+        if(dbHandler == null)
+            dbHandler = new DBHandler(context);
+
+        return dbHandler;
+    }
+
+    public Character addCharacter(Character newChar){
+        ContentValues charVals = new ContentValues();
+
+
+        charVals.put(PlayerContract.PlayerEntry.COLUMN_NAME_PLAYER_NAME, newChar.getPlayerName());
```

Comment:
You could create a method that generated a `ContentValues` from a Player. Use it here and for updating a character. Would reduce code & make your life easier w/ DB changes later on. Same comment goes for creating/updating scenarios.

Reply:
I'm slightly embarrased that I didn't think of that

Reply:
Easy to blast past stuff like this when trying to get things to work - no worries